core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -425,8 +426,26 @@ private void tryStart(Instant startedAt) {
             // For all registered output consumers, start following as close to container startup as possible
             this.logConsumers.forEach(this::followOutput);
 
+            // Wait until inspect container returns the mapped ports
+            if (exposedPorts.size() > 0) {
+                containerInfo = Unreliables.retryUntilSuccess(10, () -> {
+                    InspectContainerResponse containerInfo = dockerClient.inspectContainerCmd(containerId).exec();
+
+                    long mappedExposedPorts = containerInfo.getNetworkSettings().getPorts().getBindings().values()
+                        .stream().filter(Objects::nonNull).count();
+
+                    if (mappedExposedPorts != exposedPorts.size()) {
+                        throw new ContainerLaunchException("Inspect container response did not contain mapped ports");
```

Comment:
Should we include a small sleep to give the API results time to update?

Reply:
Oh, I thought sleeping is built into `Unreliables`, it is not?